mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -0,0 +1,144 @@
+use async_trait::async_trait;
+use common::{executor::{AbortSettings, SpawnAbortable, Timer},
+             log, Future01CompatExt};
+use futures::channel::oneshot::{self, Receiver, Sender};
+use mm2_core::mm_ctx::MmArc;
+use mm2_event_stream::{behaviour::{EventBehaviour, EventInitStatus},
+                       Event, EventStreamConfiguration};
+use mm2_number::BigDecimal;
+use std::collections::HashMap;
+
+use super::utxo_standard::UtxoStandardCoin;
+use crate::{utxo::{output_script, rpc_clients::electrum_script_hash, utxo_tx_history_v2::UtxoTxHistoryOps},
+            MarketCoinOps, MmCoin};
+
+#[async_trait]
+impl EventBehaviour for UtxoStandardCoin {
+    const EVENT_NAME: &'static str = "COIN_BALANCE";
+
+    async fn handle(self, _interval: f64, tx: oneshot::Sender<EventInitStatus>) {
+        const RECEIVER_DROPPED_MSG: &str = "Receiver is dropped, which should never happen.";
+
+        let ctx = match MmArc::from_weak(&self.as_ref().ctx) {
+            Some(ctx) => ctx,
+            None => {
+                let msg = "MM context must have been initialized already.";
+                tx.send(EventInitStatus::Failed(msg.to_owned()))
+                    .expect(RECEIVER_DROPPED_MSG);
+                panic!("{}", msg);
+            },
+        };
+
+        let addresses = match self.my_addresses().await {
```

Comment:
I assume we can have a problem here if new account/address is activated. Current implementation won't send notifications for new addresses.

Reply:
Yeah I noticed this before but forgot handling it somehow. Thanks for reminding.

Reply:
Done